File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> idx = new HashMap<>();
        int n = nums.length;

        for(int i= 0; i<n ;i++){
            if(idx.containsKey(target - nums[i])){
                return new int[] {i, idx.get(target-nums[i])};
            }
            idx.put(nums[i],i);
        }
        return new int[]{};
    }
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        idx = {}
        n = len(nums)

        for i in range(n):
            if target-nums[i] in idx:
                return [i,idx[target-nums[i]]]
            idx[nums[i]] = i
        return []
        